src/exports.cc

```diff
@@ -767,18 +767,45 @@ Word wasi_unstable_fd_fdstat_get(void *raw_context, Word fd, Word statOut) {
 }
 
 // __wasi_errno_t __wasi_environ_get(char **environ, char *environ_buf);
-Word wasi_unstable_environ_get(void *, Word, Word) {
+Word wasi_unstable_environ_get(void *raw_context, Word ptr, Word environ_buf) {
+  auto context = WASM_CONTEXT(raw_context);
+  auto &envs = context->wasm()->envs();
+  for (auto e : envs) {
+    if (!context->wasmVm()->setWord(ptr, environ_buf)) {
+      return 21; // __WASI_EFAULT
+    }
+
+    std::string data;
+    data.reserve(e.first.size() + e.second.size() + 2);
+    data.append(e.first);
+    data.push_back('=');
+    data.append(e.second);
+    data.push_back(0);
+    if (!context->wasmVm()->setMemory(environ_buf, data.size(), data.c_str())) {
+      return 21; // __WASI_EFAULT
+    }
+    environ_buf = environ_buf.u64_ + data.size();
+    ptr = ptr.u64_ + sizeof(uint32_t);
```

Comment:
yeah, we must jump by a native size though I haven't tested. But then this would not work for 32-bit Wasm VMs . Should we switch on the condition whether the caller is null vm or not (based on `context->wasmVm()->runtime()`?

Reply:
> But then this would not work for 32-bit Wasm VMs .

By native I meant native to the VM (i.e. 64-bit for NullVM and 32-bit for WasmVM). We have a `Word` type which supports that, and a ton of C++ template magic for it, but it's been years since I've looked at it, and I'm not sure if things like `ConvertWordTypeToUint32` would even work in this context.

Reply:
I don't think ConvertWordType stuff would work here. Maybe we need to introduce a template parameter on `wasi_unstable_environ_get` or just switching on runtime value which is not performant.

Reply:
Or adding a method to WasmVM like `Word incrementByPointerSize(Word)` which seems reasonable.

Reply:
This is called once at plugin startup, so I don't think it has to be super perfromant.

Maybe add `size_t WasmVm::getWordSize()` that returns either 32-bit or 64-bit, and then we can do:
```
ptr = ptr.u64_ + context->wasmVm()->getWordSize();
```
Seems like the simplest solution to me.